macosx/mac-joypad.mm

```diff
@@ -282,7 +282,9 @@ void findControls(struct JoypadDevice &device, NSDictionary *properties, NSMutab
 
     int usagePage = [properties[@kIOHIDElementUsagePageKey] intValue];
     int usage = [properties[@kIOHIDElementUsageKey] intValue];
-    if (usagePage == kHIDPage_Button)
+    if (usagePage == kHIDPage_Button ||
+        (usagePage == kHIDPage_GenericDesktop && (usage == kHIDUsage_GD_DPadUp || usage == kHIDUsage_GD_DPadDown || usage == kHIDUsage_GD_DPadLeft || usage == kHIDUsage_GD_DPadRight))
+        || (usagePage == kHIDPage_Consumer && (usage == kHIDUsage_Csmr_ACProperties || usage == kHIDUsage_Csmr_ACExit)))
```

Comment:
Adding support for kHIDUsage_GD_DPad* makes sense. SDL checks for these (and a few others besides, which I should probably add).

The kHIDUsage_Csmr*s, however, are too specific. It took me a while to figure out why your controller wanted these. If you look at the USB HID spec, these are defined as buttons for controlling GUI applications. ACProperties is "Display the properties of the current document" and ACExit is "Exit the application"

But a comment in the SDL source code says that it looks at the entire kHIDPage_Consumer "e.g. 'pause' button on Steelseries MFi gamepads."

It sucks that gamepad manufacturers are using this page, but there's not much we can do about that. But it's probably best to just check usagePage == kHIDPage_Consumer and remove the usage checks for kHIDUsage_Csmr_ACProperties and kHIDUsage_Csmr_ACExit.